File: Iniciante/1047 - Tempo de Jogo com Minutos.py
"""
Autor: Pierre Vieira
Data da submissão: 25/01/2019 20:48:37
"""
A, B, C, D = input().split(" ")
a = int(A)
b = int(B)
c = int(C)
d = int(D)
horas = c - a
minutos = d - b
if a == b == c == d:
    print("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)")
elif horas == -1: # c-a = -1
    if d > b:
        print("O JOGO DUROU 23 HORA(S) E {} MINUTO(S)".format(minutos))
    elif d < b:
        print("O JOGO DUROU 22 HORA(S) E {} MINUTO(S)".format(60 - abs(minutos)))
    else: #b = d
        print("O JOGO DUROU 23 HORA(S) E 0 MINUTO(S)")
elif a > c:
    if d > b:
        print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(24 - abs(horas), minutos))
    elif d < b:
        print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(24 - abs(horas) - 1,60 - abs(minutos)))
    else: #b = d
        print("O JOGO DUROU {} HORA(S) E 0 MINUTO(S)".format(24 - abs(horas)))
else: # c > a
    if d > b:
        print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(horas, minutos))
    elif d < b:
        print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(horas - 1,60 - abs(minutos)))
    else: #b = d
        print("O JOGO DUROU {} HORA(S) E 0 MINUTO(S)".format(horas))

File: C#/Iniciante/1047 - Tempo de Jogo com Minutos.cs
/*
* Autor: Pierre Vieira
* Data de submissão: 14/05/2020 14:24:50
*/
using System;
using System.Globalization;
using Microsoft.VisualBasic;

namespace URI {
    class URI {
        static void Main(string[] args)
        {
            string[] linha = Console.ReadLine().Split(' ');
            int a, b, c, d, instante_inicial, instante_final, duracao, duracao_horas, duracao_minutos;
            a = int.Parse(linha[0]);
            b = int.Parse(linha[1]);
            c = int.Parse(linha[2]);
            d = int.Parse(linha[3]);
            instante_inicial = a * 60 + b;
            instante_final = c * 60 + d;
            if (instante_inicial < instante_final)
            {
                duracao = instante_final - instante_inicial;
            }
            else
            {
                duracao = (24 * 60 - instante_inicial) + instante_final;
            }

            duracao_horas = duracao / 60;
            duracao_minutos = duracao % 60;

            Console.WriteLine($"O JOGO DUROU {duracao_horas} HORA(S) E {duracao_minutos} MINUTO(S)" );
        }
    }
}
